### PR TITLE
Vaccines V1

### PR DESCRIPTION
**Pull request by selvach**
## [1] AssessmentCoordinator
I seek forgiveness for putting the VaccineScreens in the assessmentCoordinator. They could potentially be split out for clarity, but we'd end up having to pass more things around. Also the COVID Tests screens are squeezed into the assessmentCoordinator. This could be refactored out - but I'm not sure it adds much....

## [2] New VaccineService & VaccineClientAPI

This current does:
 - POST to `/vaccines/` endpoint at the 
 - POST to `/dose_symptoms/` to log needle symptoms

Has associated DTO and type definitions. 

## [3] Starting Vaccine Flow

User are invited to start the VaccineFlow if the backend returns `vaccine_status` = `ask_vaccine_question`. 
The backend will return `do_not_ask` if country_code of the user is Sweden or US; or if the user is 16 or under. 

## [4] Vaccine Flow
When users start the VaccineScreens they hop though:

**VaccineYesNo Screen:**
  - Selecting Yes take you the VaccineTrialOrNational Screen
  - Selecting No exists the flow and you start your Daily Assessment. You will be asked this question EVERY DAY!

 
Screenshot 2020-12-08 at 17 21 22

 
**VaccineTrialOrNational Screen:**
  - Selecting Trail takes you to the VaccinePlaceebo Screen
  - Selecting NationalVaccie finishes the flow creates a vaccine record on the backend. Now the vaccine is done, we ask the DoseSymptoms

Screenshot 2020-12-08 at 17 21 29

**VaccinePlacebo Screen**
  - User selected Yes/No/Unsure. Always the same account -> we create a vaccing record on the backend and we go onto DoseSymptoms

Screenshot 2020-12-08 at 17 21 36

## [5] DoseSymtpoms
-  First time users see this after creating their Vaccine record on the backend. 
- It uses the same mechanism as the Daily Assessment screens (Thanks @the-kenneth !!!) and POST the payload to a new `/dose_symptoms/` backend endpoint. 
- Once done, you see the thank you page and then move onto the Daily Assessment. 

- Returning users will see this DoseSymtpoms before starting their Daily Assessment. This is controlled by the backend flag, which checks to see if they have created a vaccine / logged a dose in the last 7 days. 

Screenshot 2020-12-08 at 17 21 33

## [6] VaccineThankYou

Screenshot 2020-12-08 at 17 21 43

## [6] Daily Assessments:
Two new fields added: 
  - shiver_and_chills
   - joint_paints

## [7] Passing State

I also seek forgiveness for not using redux to pass state between the screens. (Passing vaccine_type the palcebo status).
[1] I'm a complete Redux noob
[2] This will go away in v2. (edited) 

---
## TODO:

[ ] Error handling on Server API calls / disable the buttons.

### CODE REVIEW

File: src/features/vaccines/VaccineThankYouScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+export const VaccineThankYouScreen: React.FC<Props> = ({ route, navigation }) => {
+  const handleSubmit = async () => {
+    assessmentCoordinator.gotoNextScreen(route.name);
+  };
+
+  const currentPatient = assessmentCoordinator.assessmentData.patientData.patientState;
+  return (
+    <View style={styles.rootContainer}>
+      <Screen profile={currentPatient.profile} navigation={navigation}>
```
Comment (the-kenneth) [inline src/features/vaccines/VaccineThankYouScreen.tsx:29]: At the moment I can go back and keep resubmitting a new symptoms dose. Does this matter? Maybe not? If it does we could disable the back button here?

Comment (selvach) [reply src/features/vaccines/VaccineThankYouScreen.tsx:29]: Yes! Agreed!!

Comment (the-kenneth) [reply src/features/vaccines/VaccineThankYouScreen.tsx:29]: Theres a flag to disable it on Screen

Comment (selvach) [reply src/features/vaccines/VaccineThankYouScreen.tsx]: Updated!

**Review by the-kenneth — APPROVED** [review]

**Pull request merged**